server/web/api/register.py

```diff
@@ -69,58 +66,51 @@ def login():
 @api.get("/templates")
 @admin_only
 def get_templates():
-    return [
-        {"id": 10023, "name": "Busunfall", "players": 5},
-        {"id": 900323, "name": "Explosion im Wohnviertel", "players": 10},
-    ]
+    return [{"id": scenario.id, "name": scenario.name, "executions": [execution.id for execution in scenario.executions]}
+            for scenario in models.Scenario.query]
 
 
-@api.post("/scenario/start")
+@api.post("/scenario")
+@required("id", int, RequiredValueSource.FORM)
 @admin_only
-def start_scenario():
-    try:
-        id = request.form["id"]
-    except KeyError:
-        return {"error": "Missing id in request"}, 400
-
-    # TODO: Create actual execution
-    exec_id = math.floor(random() * 1000)
-    test_execution.update({
-        "id": exec_id,
-        "status": "",
-        "players": [
-            {
-                "tan": x,
-                "name": "Max Mustermann",
-                "status": choice(["", "In Vorbereitung"]),
-                "action": "Legt Zugang",
-            }
-            for x in [str(x) for x in uniques(5)]
-        ],
-    })
-    return {
-        "id": exec_id,
-    }
+def start_scenario(id: int):
+    if load_execution(id):
+        return active_executions[id].to_dict()
 
+    # Failure
+    raise NotFound(f"Execution with id={id} does not exist")
 
-@api.get("/execution/<int:id>")
-@admin_only
+
+@api.get("/execution")
+@required("id", int, RequiredValueSource.ARGS)
+# @admin_only
+@csrf.exempt
 def get_execution_status(id: int):
-    test_execution["id"] = id
-    random_player = choice(test_execution["players"])
-    random_player["status"] = "In Vorbereitung" if random_player["status"] == "" else ""
-    return test_execution
+    execution = try_get_execution(id)
+    return execution.to_dict()
 
 
-@api.post("/execution/<int:id>/start")
+@api.patch("/execution")
+@required("id", int, RequiredValueSource.ARGS)
+@required("running", booleanize, RequiredValueSource.FORM)
 @admin_only
-def start_execution(id: int):
-    test_execution["status"] = "Aktiv"
-    return test_execution
+def change_execution_status(id: int, running: bool):
+    execution = try_get_execution(id)
+    execution.status = Execution.Status.RUNNING if not running else Execution.Status.PENDING
+    return Response(status=200)
```

Comment:
Isn't `Execution.Status.FINISHED` missing here?

Reply:
I have changed the button to a dropdown to make it possible to set the status to anything you want (except unknown).

Reply:
Wouldn't it be nicer, if there is a start button for pending executions and the start button turns into a stop button when the execution is running? Just a suggestion.

Reply:
@simohlsen I totally agree with you. However, this should probably be part of a future PR.